qiskit/pulse/instructions/acquire.py

```diff
@@ -0,0 +1,235 @@
+
+        if channels is not None:
+            super().__init__(duration, *channels, *mem_slot, *reg_slot, name=name)
+        else:
+            super().__init__(duration, name=name)
+
+        self._acquires = channels
+        self._channel = channels[0] if channels else None
+        self._mem_slots = mem_slot
+        self._reg_slots = reg_slot
+        self._kernel = kernel
+        self._discriminator = discriminator
+
+    @property
+    def operands(self) -> List:
+        """Return a list of instruction operands."""
+        return [self.duration, self.channel,
+                self.mem_slot, self.reg_slot,
+                self.kernel, self.discriminator]
+
+    @property
+    def channel(self) -> AcquireChannel:
+        """Return the :py:class:`~qiskit.pulse.channels.Channel` that this instruction is
+        scheduled on.
+        """
+        return self._channel
+
+    @property
+    def kernel(self) -> Kernel:
+        """Return kernel settings."""
+        return self._kernel
+
+    @property
+    def discriminator(self) -> Discriminator:
+        """Return discrimination settings."""
+        return self._discriminator
+
+    @property
+    def acquire(self) -> AcquireChannel:
+        """Acquire channel to acquire data. The ``AcquireChannel`` index maps trivially to
+        qubit index.
+        """
+        return self._acquires[0] if self._acquires else None
+
+    @property
+    def mem_slot(self) -> MemorySlot:
+        """The classical memory slot which will store the classified readout result."""
+        return self._mem_slots[0] if self._mem_slots else None
+
+    @property
+    def reg_slot(self) -> RegisterSlot:
+        """The fast-access register slot which will store the classified readout result for
+        fast-feedback computation.
+        """
+        return self._reg_slots[0] if self._reg_slots else None
+
+    @property
+    def acquires(self) -> List[AcquireChannel]:
+        """Acquire channels to be acquired on."""
+        return self._acquires
+
+    @property
+    def mem_slots(self) -> List[MemorySlot]:
+        """MemorySlots."""
+        return self._mem_slots
+
+    @property
+    def reg_slots(self) -> List[RegisterSlot]:
+        """RegisterSlots."""
+        return self._reg_slots
+
+    def __repr__(self):
+        return "{}({}{}{}{})".format(self.__class__.__name__,
+                             self.duration,
+                             ', ' + ', '.join(str(ch) for ch in self.acquires) if self.acquires else '',
+                             ', ' + str(self.mem_slot) if self.mem_slot else '',
+                             ', ' + str(self.reg_slot) if self.reg_slot else '',
+                             ', ' + str(self.kernel) if self.kernel else '',
+                             ', ' + str(self.discriminator) if self.discriminator else '')
+
+    def __eq__(self, other):
+        return self.operands == other.operands
```

Comment:
Won't this give equality even if the instruction types are different?

Reply:
I'll update, thank you! acquire's operands are very specific to acquire, but that is only true now, and not as robust as also checking the type